tests/check_flatpak_extra_languages.pm

```diff
@@ -0,0 +1,32 @@
+# vi: set shiftwidth=4 tabstop=4 expandtab:
+use base 'installedtest';
+use strict;
+use testapi;
+use utils;
+
+sub run {
+    my $self = shift;
+    my $default_extra_languages = "ar;bn;en;es;fr;id;pt;th;vi;zh";
+    my $extra_languages = get_var('EOS_IMAGE_FLATPAK_LOCALES', '');
+    $extra_languages =~ s/\ /;/ig;
+
+    # Test extra-languages key is present
+    $self->user_console();
+    if ($extra_languages eq $default_extra_languages) {
+        assert_script_run('flatpak config --system --get extra-languages', 60, '*unset*');
+    } elsif (!$extra_languages) {
+        assert_script_run('flatpak config --system --get extra-languages', 60, $extra_languages);
+    }
+
+    $self->exit_user_console();
```

Comment:
This console-based testing looks good, but the [testing you were doing manually](https://phabricator.endlessm.com/T24214#777438) also checked that apps from the extra locales were shown in gnome-software’s UI. That’s the user-visible behaviour that we want, so that’s what should be tested in OpenQA (in addition to this console-based stuff).

Perhaps a robust way to test it would be to do the console-based testing above, and then call `flatpak config --system --set extra-languages "some known value"`, then run gnome-software and compare its main list to a new needle which shows apps from the languages in the new `"some known value"` list of locales. You’d choose the list of locales to make the gnome-software main window obviously differ from its normal app list.

Reply:
Aha, that's a good idea - will do that